controllers/helpers.go

```diff
@@ -601,10 +603,21 @@ func GetClusterIdentityFromRef(ctx context.Context, c client.Client, azureCluste
 	return nil, nil
 }
 
-func clusterIdentityFinalizer(prefix, clusterNamespace, clusterName string) string {
+// deprecatedClusterIdentityFinalizer is was briefly used to compute a finalizer without a hash in releases v1.5.1 and v1.4.4.
+// It is kept here to ensure that we can remove it from existing clusters for backwards compatibility.
+// This function should be removed in a future release.
+func deprecatedClusterIdentityFinalizer(prefix, clusterNamespace, clusterName string) string {
 	return fmt.Sprintf("%s/%s-%s", prefix, clusterNamespace, clusterName)
 }
 
+// clusterIdentityFinalizer generates a finalizer key.
+// The finalizer key is a combination of the prefix and a hash of the cluster name and namespace.
+// We use a hash to ensure that the finalizer key name is not longer than 63 characters.
+func clusterIdentityFinalizer(prefix, clusterNamespace, clusterName string) string {
+	hash := sha256.Sum224([]byte(fmt.Sprintf("%s-%s", clusterNamespace, clusterName)))
```

Comment:
looked at which hashing algorithm to use and settled on sum 224 because: 
- sha-1 and md5 are flagged by the linter as blocklisted (too weak)
- sha-256 is too long (64 characters)

If anyone has any suggestions for a better crypto package to use please speak up!

Reply:
IMO So long as we have deterministic uniqueness we're good (the linter should be smarter than this, as we aren't actually doing any secret encryption, but I digress...)